testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/flows/AbstractOIDCResponseTypeTest.java

```diff
@@ -253,6 +253,8 @@ private void oidcFlow(String expectedAccessAlg, String expectedIdTokenAlg) throw
         for (IDToken idt : idTokens) {
             Assert.assertEquals("abcdef123456", idt.getNonce());
             Assert.assertEquals(authzResponse.getSessionState(), idt.getSessionState());
+            // see KEYCLOAK-9635
+            Assert.assertNotNull("claim at_hash should be present in IDToken for OIDC auth code flow requests", idt.getAccessTokenHash());
```

Comment:
One could optionally verify the generated hash, but I did not find a method to do that. Would be great if someone could point me to a method where the at_hash claim is checked...? :)

Reply:
@thomasdarimont It is done for example in testAuthzResponseAndRetrieveIDTokens() of various tests, which already generate "at_has" . For example in OIDCImplicitResponseTypeIDTokenTokenTest.testAuthzResponseAndRetrieveIDTokens() .

I suggest to explicitly add the verifications to the individual methods to explicitly check OIDCXYTest.testAuthzResponseAndRetrieveIDTokens() to explicitly check if "at_hash" is generated always when it makes sense instead of using it here in the subclass where the check is not very explicit. WDYT?